packages/user/FractalCore/ui/src/components/modals/set-guild-rep-modal.tsx

```diff
@@ -0,0 +1,82 @@
+import { useEffect } from "react";
+import { z } from "zod";
+import { useGuildAccount } from "@/hooks/use-guild-account";
+import {
+    Dialog,
+    DialogContent,
+    DialogHeader,
+    DialogTitle,
+} from "@shared/shadcn/ui/dialog";
+import { useAppForm } from "@shared/components/form/app-form";
+import { useSetGuildRep } from "@/hooks/fractals/use-set-guild-rep";
+import { FieldAccountExisting } from "@shared/components/form/field-account-existing";
+import { supervisor } from "@/supervisor";
+
+export const SetGuildRepModal = ({
+    show,
+    openChange,
+}: {
+    show: boolean;
+    openChange: (show: boolean) => void;
+}) => {
+    const { mutateAsync: setGuildRep } = useSetGuildRep();
+    const guildAccount = useGuildAccount();
+
+    const form = useAppForm({
+        defaultValues: {
+            representation: {
+                newRep: ""
+            },
+        },
+        onSubmit: async ({ value: { representation: { newRep } } }) => {
+            await setGuildRep([guildAccount!, newRep]);
+            openChange(false);
+        },
```

Comment:
I wonder if we could reset the form in this `onSubmit` instead of the `useEffect` on `L44` below. If so, that could be applied to all of these modals. Would something like this work?

```
onSubmit: async ({ value: { representation: { newRep } }, formApi }) => {
    await setGuildRep([guildAccount!, newRep]);
    openChange(false);
    formApi.reset();
},
```